interfaces/systemd/backend.go

```diff
@@ -69,16 +74,22 @@ func (b *Backend) Setup(snapInfo *snap.Info, confinement interfaces.ConfinementO
 	}
 	glob := interfaces.InterfaceServiceName(snapName, "*")
 
-	systemd := sysd.New(dirs.GlobalRootDir, sysd.SystemMode, &dummyReporter{})
+	var systemd sysd.Systemd
+	if b.preseed {
+		systemd = sysd.NewEmulationMode(dirs.GlobalRootDir)
+	} else {
+		systemd = sysd.New(dirs.GlobalRootDir, sysd.SystemMode, &dummyReporter{})
+	}
+
 	// We need to be carefully here and stop all removed service units before
 	// we remove their files as otherwise systemd is not able to disable/stop
 	// them anymore.
-	if err := disableRemovedServices(systemd, dir, glob, content); err != nil {
+	if err := b.disableRemovedServices(systemd, dir, glob, content); err != nil {
 		logger.Noticef("cannot stop removed services: %s", err)
 	}
 	changed, removed, errEnsure := osutil.EnsureDirState(dir, glob, content)
 	// Reload systemd whenever something is added or removed
-	if len(changed) > 0 || len(removed) > 0 {
+	if (len(changed) > 0 || len(removed) > 0) && !b.preseed {
```

Comment:
I was expecting b.preseed to be only used to instantiate emulation or non-emulation systemd object, this would be less error prone than the current logic.

Reply:
As a secondary nitpick, I would prefer to move the condition `!b.preseed` to the front of each conditional bits as an alternative to the probably more invasive change I outlined above.

Reply:
the nitpick about checking !b.preseed always first makes sense to me